lib/UR/Object/Type/InternalAPI.pm

```diff
@@ -910,7 +921,7 @@ sub most_specific_subclass_with_table {
 
     foreach my $class_name ( $self->class_name->inheritance ) {
         my $class_obj = UR::Object::Type->get(class_name => $class_name);
-        return $class_name if ($class_obj && $class_obj->table_name);
+        return 1 if ($class_obj and $class_obj->has_direct_table);
```

Comment:
That would be a bug.  I've pushed another commit to fix it.

BTW, most_general_subclass_with_table() was also changed in a similar way bot didn't have this bug.